source/Octopus.Tentacle/Communications/HalibutInitializer.cs

```diff
@@ -100,7 +102,19 @@ void AddPollingEndpoints()
 
                 var halibutTimeoutsAndLimits = new HalibutTimeoutsAndLimits();
                 var serviceEndPoint = new ServiceEndPoint(pollingEndPoint.Address, pollingEndPoint.Thumbprint, halibutProxy, halibutTimeoutsAndLimits);
-                halibut.Poll(new Uri(pollingEndPoint.SubscriptionId), serviceEndPoint, CancellationToken.None);
+
+                //Open multiple polling connections if the env var is set to a non-zero/negative number
+                var connectionCount = int.TryParse(Environment.GetEnvironmentVariable(EnvironmentVariables.TentaclePollingConnectionCount), out var count)
+                    ? count
+                    : 1;
+
+                // negative numbers and 0 are coerced to 1
+                connectionCount = Math.Max(connectionCount, 1);
```

Comment:
It might be worth adding a hard limit here, e.g. 8 or 16 connections.  Just trying to make sure that people won't shoot themselves in the foot. We can always relax this limit  if there is a customer with a reasonable use case. 

Also, it would be worth logging what the effective `connectionCount`  is, especially if we apply a hard limit.

Reply:
yeah, I did think about that. I wasn't sure. Happy to set it to `8` for now and log it out 👍

Reply:
@pawelpabich Implemented a hard limit of 8 and also log the requested and resulting value (after validation/coercion) in [f251b2b](https://github.com/OctopusDeploy/OctopusTentacle/pull/881/commits/f251b2b48defb7687450d602d93e9641a73022e5)